### ISSUE TITLE
[FEATURE]: Clicking the "Play/Pause" button when "Select Staff" page is openning

### ISSUE THREAD

**Issue by chunyinng078** [COMP3122-2324-Project/Simpsyber#11]
In the "Select Staff" page, players/users cannot click the "Play/Pause" button to start/pause the game directly. Players/users have to close the current window first. It is very inconvenient.

Deciding to assign staff to tasks takes time; if players/users enter the page without pausing the game, time might be wasted. And, when players/users notice that the timer is still running when they choose staff, they need to close the "Select Staff" window first, which would also cause time wasted.

https://github.com/COMP3122-2324-Project/Simpsyber/assets/95467201/b71a1549-93ba-4c63-870a-5eadb895059d

This enhancement would be helpful for users/players and improve the user experience. Therefore, I suggest an enhancement that players/users can click the "Play/Pause" button while the "Select Staff" page is opening. So that users/players can stop the timer when selecting staff for the task, which helps users/players utilize their time.

I consider the issue is a feature but not a defect because I think the previous design should designed to not allow player to click on the button when the tab is opening.

**Comment by Generic626** [comment]
First I totally agree with your suggestion. Great !

But to build upon/ elevate this, I would suggest whenever the user opens the "Select Staff" menu, the timer would automatically stop by itself. That way the user won't need to do another step which is to press the pause button.

**Comment by chunyinng078** [comment]
@Generic626  Automatically stopping the timer may be a good solution! If the timer stops itself, there is no redundancy action regarding controlling the game timer. Users could be more focused on what they would be doing, like assigning tasks, which could provide a better user experience.

Based on my issue regrading the "ease of timer controlling" and to elevate your suggestion, I would also suggest when the users close the "Select Staff" menu, and the timer will continue automatically. The flow would be seamless, making the function of timer control easier to use, providing an even better user experience!

**Comment by LeungYatOn** [comment]
That's true! Even the x1 speed is too faster for me to operate. I have a better idea for the issue. How about to timer will automatically stop whenever someone was completed their own task. 

For example, after you assigned the 2 satff to work on the tasks. When 1 of them completed the task, the timer will automatically stop so that I can assign him to work immediately. I will not waste any human resource.

**Comment by LawPakSum** [comment]
I agree with you. In my opinion, there would be a lot of time wasted If we need to pause the game by ourselves. Besides selecting staff would active the pause function, may be users can select if pause the time immediately or not, after a task is finished by a staff. Since most of the time, we would assign the staff to do another job after his/her task is finished. And then we need to do the extra effort to pause the name. If we are using x2 speed, there could be many time to be wasted. This kind of extra operation sometimes makes me feels ruined the game and user unfriendly.
